frontend/src/components/SourceHistoryListItem/SourceHistoryListItem.tsx

```diff
@@ -32,7 +33,7 @@ export const SourceHistoryListItem: FunctionComponent<ComponentProps> = (props)
         })
         .catch((error) => console.log(error.message));
     }
-  }, [fetching]);
+  }, [fetching, item]);
```

Comment:
With this one too.

Reply:
It seems not to work for data sources though

Reply:
I'm hoping the new approach would eliminate the need for this change. If not, don't mind it.